.upstream-tests/test/cuda/proclaim_return_type.pass.cpp

```diff
@@ -60,32 +62,42 @@ struct d_callable
 
 int main(int argc, char ** argv)
 {
-#ifdef __CUDA_ARCH__
-#define TEST_SPECIFIER(...)                                                    \
-  {                                                                            \
-    test_proclaim_return_type<double>([] __VA_ARGS__ () { return 42.0; },      \
-                                      42.0);                                   \
-    test_proclaim_return_type<int>([] __VA_ARGS__ (int v) { return v * 2; },   \
-                                   42, 21);                                    \
-                                                                               \
-    int v = 42;                                                                \
-    int* vp = &v;                                                              \
-    test_proclaim_return_type<int&>(                                           \
-        [vp] __VA_ARGS__ () -> int& { return *vp; }, v);                       \
-  }
+  int v = 42;
+  int* vp = &v;
+
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (
+    test_proclaim_return_type<int>(hd_callable{}, 42);
+    test_proclaim_return_type<int>(d_callable{}, 42);
+  ),(
+    test_proclaim_return_type<double>([]   { return 42.0; }, 42.0);
+    test_proclaim_return_type<int>   ([]   (const int v) { return v * 2; }, 42, 21);
+    test_proclaim_return_type<int&>  ([vp] () -> int& { return *vp; }, v);
+
+    test_proclaim_return_type<int>(hd_callable{}, 42);
+  ))
+
+  // execution space annotations on lambda require --extended-lambda flag with nvrtc
+#if !defined(TEST_COMPILER_NVRTC)
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (
+    test_proclaim_return_type<double>([]   __device__ { return 42.0; }, 42.0);
+    test_proclaim_return_type<int>   ([]   __device__ (const int v) { return v * 2; }, 42, 21);
+    test_proclaim_return_type<int&>  ([vp] __device__ () -> int& { return *vp; }, v);
+
+    test_proclaim_return_type<double>([]   __host__ __device__ { return 42.0; }, 42.0);
+    test_proclaim_return_type<int>   ([]   __host__ __device__ (const int v) { return v * 2; }, 42, 21);
+    test_proclaim_return_type<int&>  ([vp] __host__ __device__ () -> int& { return *vp; }, v);
+  ),(
+    test_proclaim_return_type<int>(h_callable{}, 42);
+  ))
```

Comment:
```suggestion
  test_proclaim_return_type<double>([] __host__ __device__ { return 42.0; }, 42.0);
  test_proclaim_return_type<int>   ([] __host__ __device__ (const int v) { return v * 2; }, 42, 21);

  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (
    test_proclaim_return_type<double>([]   __device__ { return 42.0; }, 42.0);
    test_proclaim_return_type<int>   ([]   __device__ (const int v) { return v * 2; }, 42, 21);
    test_proclaim_return_type<int&>  ([vp] __device__ () -> int& { return *vp; }, v);

    test_proclaim_return_type<int&>  ([vp] __host__ __device__ () -> int& { return *vp; }, v);
  ),(
    test_proclaim_return_type<int>(h_callable{}, 42);
  ))
```

Reply:
This is not possible, we get an error from inside the implementation:
```
nvcc_internal_extended_lambda_implementation:310:22: error: definition of implicit copy constructor for '__nv_hdl_wrapper_t<false, true, false, __nv_dl_tag<int (*)(int, char **), &fake_main, 1>, double ()>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
__nv_hdl_wrapper_t & operator=(const __nv_hdl_wrapper_t &in) = delete;
```